packages/react-components/src/components/RichTextEditor/RichTextEditor.tsx

```diff
@@ -215,9 +235,13 @@ export const RichTextEditor = React.forwardRef<RichTextEditorComponentRef, RichT
         /* @conditional-compile-remove(rich-text-editor-image-upload) */
         let removedInlineImages: Record<string, string>[] = [];
         /* @conditional-compile-remove(rich-text-editor-image-upload) */
-        if (shouldUpdateInlineImages) {
+        if (shouldRemoveInlineImages) {
           /* @conditional-compile-remove(rich-text-editor-image-upload) */
           removedInlineImages = getRemovedInlineImages(content, previousInlineImages);
+          /* @conditional-compile-remove(rich-text-editor-image-upload) */
+          if (removedInlineImages.length > 0) {
+            removeLocalBlobs(inlineImageLocalBlobs, removedInlineImages);
+          }
```

Comment:
Minor: Do we need the two CC statements below if we have one for parent if?
```suggestion
        /* @conditional-compile-remove(rich-text-editor-image-upload) */
        if (shouldRemoveInlineImages) {
          removedInlineImages = getRemovedInlineImages(content, previousInlineImages);
          if (removedInlineImages.length > 0) {
            removeLocalBlobs(inlineImageLocalBlobs, removedInlineImages);
          }
```